app/commands/git/sync_exercise.rb

```diff
@@ -0,0 +1,92 @@
+module Git
+  class SyncExercise
+    include Mandate
+    initialize_with :exercise
+
+    def call
+      lookup_head_and_current_commit
+      update_exercise! unless exercise_synced_to_head?
+    end
+
+    private
+    attr_reader :current_commit, :head_commit
+
+    def lookup_head_and_current_commit
+      exercise.git.update!
+
+      @current_commit = exercise.git.lookup_commit(exercise.synced_to_git_sha)
+      @head_commit = exercise.git.head_commit
+    end
+
+    def exercise_synced_to_head?
+      current_commit.oid == head_commit.oid
+    end
+
+    def update_exercise!
+      update_concept_exercise! if exercise.concept_exercise?
+      update_practice_exercise! if exercise.practice_exercise?
```

Comment:
This is a good example of where you can use `send`:
```suggestion
      send("update_#{exercise.git_type}!")
```